Domain/ProductManagement/InventoryUI.cs

```diff
@@ -0,0 +1,181 @@
+            {
+                PrintMessage(ex.Message);
+            }
+        }
+
+        Inventory.MainMenu();
+    }
+
+    public void UpdateProductUI()
+    {
+        var productName = Input("Enter the product name: ");
+        DisplayProductUpdateMenu();
+        int choice = int.Parse(UserSelection());
+
+        try
+        {
+            Inventory.UpdateProduct(productName, choice);
+            PrintMessage("Product updated successfully.");
+        }
+        catch (Exception ex)
+        {
+            PrintMessage(ex.Message);
+        }
+        
+        Inventory.MainMenu();
+    }
+
+    public void UpdateProductNameUI()
+    {
+        var name = Input("Enter the product name you want to change: ");
+        var newName = Input("Enter the new name: ");
+
+        var product = Inventory.GetProductByName(name);  
+        if (product != null)
+        {
+            Inventory.UpdateProductName(product, newName);  
+            PrintMessage("Product name updated successfully.");
+        }
+        else
+            PrintMessage("Product not found.");
+        
+        Inventory.MainMenu();
+    }
+
+    public void UpdateProductPriceUI()
+    {
+        var name = Input("Enter the product name you want to change its price: ");
+        var newPrice = double.Parse(Input("Enter the new price: "));
+
+        var product = Inventory.GetProductByName(name);
+        if (product != null)
+        {
+            Inventory.UpdateProductPrice(product, newPrice);
+            PrintMessage("Product price updated successfully.");
+        }
+        else
+            PrintMessage("Product not found.");
+
+        Inventory.MainMenu();
+    }
+
+    public void UpdateProductQuantityUI()
+    {
+        var name = Input("Enter the product name you want to change its quantity: ");
+        var newQuantity = int.Parse(Input("Enter the new quantity: "));
+
+        var product = Inventory.GetProductByName(name);
+        if (product != null)
+        {
+            Inventory.UpdateProductQuantity(product, newQuantity);
+            PrintMessage("Product quantity updated successfully.");
+        }
+        else
+            PrintMessage("Product not found.");
+
+        Inventory.MainMenu();
+    }
+
+    public void DisplayProductUpdateMenu()
+    {
+        Console.WriteLine(" -------------------------------- ");
+        Console.WriteLine("| What do you want to update?    |");
+        Console.WriteLine(" -------------------------------- ");
+        Console.WriteLine("| 1: Update the product name     |");
+        Console.WriteLine("| 2: Update the product price    |");
+        Console.WriteLine("| 3: Update the product quantity |");
+        Console.WriteLine("| 4: Exit to the main menu       |");
+        Console.WriteLine(" -------------------------------- ");
+    }
+
+    public void ViewAProductUI()
+    {
+        var noOfProducts = int.Parse(Input("How many products do you want to view their details?"));
```

Comment:
if the input is not an integer or an empty string, this will throw an exception. Let handle this type of errors.